bot/exts/help_channels/_channel.py

```diff
@@ -109,11 +118,21 @@ async def help_thread_opened(opened_thread: discord.Thread, *, reopen: bool = Fa
         await _close_help_thread(opened_thread, _stats.ClosingReason.CLEANUP)
         return
 
+    # Discord sends the open event long before the thread is ready for actions in the API.
+    # This causes actions such as fetching the message, pinning message, etc to fail.
+    # We sleep here to try and delay our code enough so the thread is ready in the API.
+    await asyncio.sleep(2)
```

Comment:
Are we sure that 2 seconds will always be enough ? Isn't there another event we can subscribe to, like the starter message for example ?

Reply:
Yea, theoretically we could `wait_for` an `on_message` event which has the same message_id as the just created thread. Since this would then mean the message was sent to us via the gateway, and is also cached in d.py